pgx-pg-sys/build.rs

```diff
@@ -536,14 +536,26 @@ fn run_bindgen(pg_config: &PgConfig, include_h: &PathBuf) -> eyre::Result<syn::F
         .header(include_h.display().to_string())
         .clang_arg(&format!("-I{}", includedir_server.display()))
         .parse_callbacks(Box::new(PgxOverrides::default()))
+        .allowlist_file(".*confdefs.h")
+        .allowlist_file(".*([fF]uncs|mgr).h")
+        .allowlist_file(".*htup(_details)?.h")
+        .allowlist_file(".*syscache.h")
+        .allowlist_file(".*mcxt.h")
+        .allowlist_file(".*(storage|catalog|access|commands|executor|adt|optimizer|rewrite|postmaster|tcop|replication|nodes|postgres|parse|pg_|item|heap).*")
+        .allowlist_function("memcpy")
+        .allowlist_var("SIG.*")
         .blocklist_type("Datum") // manually wrapping datum types for correctness
         .blocklist_type("NullableDatum")
         .blocklist_function("varsize_any") // pgx converts the VARSIZE_ANY macro, so we don't want to also have this function, which is in heaptuple.c
         .blocklist_function("query_tree_walker")
         .blocklist_function("expression_tree_walker")
-        .blocklist_function("sigsetjmp")
-        .blocklist_function("siglongjmp")
+        .blocklist_function(".*setjmp")
+        .blocklist_function(".*longjmp")
         .blocklist_function("pg_re_throw")
+        .blocklist_item("CONFIGURE_ARGS") // configuration during build is hopefully irrelevant
+        .blocklist_item("_*(HAVE|have)_.*") // these are for C's use
+        .blocklist_item("__[A-Z].*") // these are reserved
+        .blocklist_function("float[48].*") // Rust has plenty of float handling
```

Comment:
As far as I can tell, we bind a lot of functions that are actually meant to be called into by Postgres only to handle SQL runtime requests, and thus are of no interest for PGX users because they can just use Rust code to handle them. This is a fairly conservative start on omitting those functions because I deemed floats as least likely to be needed to be handled by C (and Rust passing floats to C can, in rare cases, on architectures we currently don't intentionally support, actually lead to busted behavior).